agent.py

```
@@ -0,0 +1,1114 @@
+        elif mem_str.endswith('Mi'):
+            return int(mem_str[:-2])
+        elif mem_str.endswith('M'):
+            return int(mem_str[:-1])
+        else:
+            return int(mem_str)  # assume bytes, but rare; fallback to MiB estimation
+
+    def _modify_pod_manifest(self, pod_name: str, namespace: str, container_name: str) -> tuple:
+        try:
+            # Step 1: Fetch pod manifest
+            cmd = f"kubectl get pod {pod_name} -n {namespace} -o yaml"
+            result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
```

Ruff (0.13.1)

`subprocess` call with `shell=True` identified, security issue

(S602)